tools/linker/MobileExtensions.cs

```diff
@@ -37,6 +50,48 @@ static bool HasGeneratedCodeAttribute (ICustomAttributeProvider provider, Derive
 			return provider.HasCustomAttribute (context, "System.Runtime.CompilerServices", "CompilerGeneratedAttribute");
 		}
 
+		// The 'provider' parameter is only used in error messages to explain where the broken attribute comes from
+		// (in particular it's not used to get the custom attributes themselves, since those may not come from this provider instance)
+		static BindingImplOptions? GetBindingImplAttribute (ICustomAttributeProvider provider, IEnumerable<ICustomAttribute> attributes)
+		{
+			if (attributes == null)
+				return null;
+			
```

Comment:
The compiler will do that for you if you want:

```
foreach (var ca in attributes ?? Enumerable.Empty< ICustomAttribute>()) {
        // code here
}
```
Looking at the IL, it will do the null check, then add the loop. So AFAIK you will not need to do the null check. Again, if you want to. It is not a real issue.

Reply:
Just playing the devil's advocate here but `Enumerable.Empty< ICustomAttribute>()` is more expensive than a `null` check 😝

Reply:
That's not a bad idea, but this time I think the code reads easier with the null check (it's a very standard code sequence, and easy to read).